src/modules/mana/selectors.ts

```diff
@@ -76,3 +85,58 @@ export const getWalletWithdrawals = createSelector<
 >(getWithdrawals, getAddress, (withdrawals, address) =>
   address ? withdrawals.filter((withdrawal) => withdrawal.from === address) : []
 )
+
+export const getTransactionByNetwork = createSelector<
+  RootState,
+  Transaction[],
+  Deposit[],
+  Withdrawal[],
+  Record<Network, AccountTransaction[]>
+>(
+  getTransactionsData,
+  getDeposits,
+  getWithdrawals,
+  (transactions, deposits, withdrawals) => {
+    const result: Record<Network, AccountTransaction[]> = {
+      ETHEREUM: [],
+      MATIC: [],
+    }
+    for (const tx of transactions) {
+      const { network }: { network: Network } = getChainConfiguration(
+        tx.chainId
+      )
+      const deposit = deposits.find((deposit) => tx.hash === deposit.hash)
+      const withdrawal = withdrawals.find(
+        (withdrawal) => tx.hash === withdrawal.hash
+      )
+      if (deposit) {
+        const accountTransaction: AccountTransaction<Deposit> = {
+          hash: tx.hash,
+          type: TransactionType.DEPOSIT,
+          status: mapStatus(tx.status),
+          data: deposit,
+        }
+        result[network].push(accountTransaction)
+      } else if (withdrawal) {
+        const accountTransaction: AccountTransaction<Withdrawal> = {
+          hash: tx.hash,
+          type: TransactionType.WITHDRAWAL,
+          status: mapStatusWithdrawal(withdrawal.status),
+          data: withdrawal,
+        }
+        result[network].push(accountTransaction)
+      } else {
+        if (tx.actionType === SEND_MANA_SUCCESS) {
+          const accountTransaction: AccountTransaction<Send> = {
+            hash: tx.hash,
+            type: TransactionType.SEND,
+            status: mapStatus(tx.status),
+            data: tx.payload.send || {},
```

Comment:
There's no `send` in the payload of `SendManaSuccessAction`: https://github.com/decentraland/account/blob/master/src/modules/mana/actions.ts#L122

There's `amount`, `to` and `network`.

You don't only need to filter the tx by `actionType` but make sure the `tx.from` equals the conencted wallet address. You can get the connected wallet address by using the `getAddress` selector from `decentraland-dapps` and include it in this selector's dependencies. Otherwise when you switch accounts you will see the history from a different account.

Reply:
There is a `send` in the payload here:
https://github.com/decentraland/account/pull/90/files#diff-1191a12c7a0d818085dc071442b51a6074acd6d0095feab0673d8ccdaacba346R119

Thanks! I'll add the wallet check too.